app/models/comment.rb

```diff
@@ -466,20 +477,31 @@ def count_visible_comments
   end
 
   def check_for_spam?
-    #don't check for spam while running tests or if the comment is 'signed'
-    self.approved = Rails.env.test? || !self.pseud_id.nil? || !Akismetor.spam?(akismet_attributes)
+    is_logged_in = !self.pseud_id.nil? 
+    is_account_old_enough = is_logged_in && !self.pseud.user.should_spam_check_comments?
+    is_on_tag = self.on_tag?
+    
+    should_skip_checking = is_logged_in && (is_account_old_enough || is_on_tag)
+
+    self.approved = should_skip_checking || !Akismet.spam?(akismet_attributes)
+  end
+
+  def submit_spam
```

Comment:
Would it be doable to override these methods in the tests instead of the ones on Akismet? I'm a bit worried about adding an extra abstraction layer (the Akismet module) just to be able to stub methods in tests .